### ISSUE TITLE
Slotted children reordering 

### ISSUE THREAD

**Issue by schadenn** [ionic-team/stencil#1725]
**Stencil version:**
```
 @stencil/core@1.1.6
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

I've implemented a select dropdown element that can also show an error-text when the selection is invalid. Everytime I set/reset the error-text on the <my-select> element, the options within the select (passed as slotted children) will reorder.

**Expected behavior:**

The children shall be in the same order as before.

**Steps to reproduce:**

Here is a github reproduction repo:
https://github.com/schadenn/testing-stencil-slotted-children-bug

When you start it you'll see a button that opens a select dropdown. Keep the dropdown open, choose the <lsg-select> element in Chrome debugger and set/reset error-text $0.setAttribute("error-text", "Test")/$0.setAttribute("error-text", "").

**Related code:**
See repo above

**Other information:**
Could be that this only happens for components not using shadow-dom.

**Comment by manucorporat** [comment]
Could you test using the latest version of stencil?

**Comment by schadenn** [comment]
Did. Still the same in 1.1.6.

**Comment by schadenn** [comment]
We're experiencing this issue alot. Mainly using slotted components and polyfills because we need to support IE. There is alot of "not keeping the order" going on when you do conditional renders/renders based on props. Even if I set keys.

**Comment by ryanhaney** [comment]
I believe the problem resides in placing `<slot/>` within `<Host>` elements. @schadenn, can you try to wrap the slot with a div and see if that works?

**Comment by schadenn** [comment]
@ryanhaney @manucorporat 
Interesting. This suggestion fixes it. Any insights why (guess something with stencil losing reference of the DOM elements)? :)

**Comment by ryanhaney** [comment]
My issue was more like this:

definition:
```
<Host>
  <slot/>
</Host>
```

usage:
```
<some-component>
  <div>Child 1</div>
  <div>Child 2</div>
  <div>Child 3</div>
  <div>Child 4</div>
</some-component>
```

As written, `some-component` will render its slotted children out-of-order. When I changed the definition to:

```
<Host>
  <div>
    <slot/>
  </div>
</Host>
```
... some-component rendered its children as expected.

**Comment by ryanhaney** [comment]
I also found that if the usage was placed directly in an HTML file, the ordering was fine. The issue presented itself when the usage of `some-component` was nested deeper in the app tree.

**Comment by splitinfinities** [comment]
Hey there, thank you for the patience getting back to you. The new team is getting started and we're working through the backlog now. 

We do not support Stencil versions less than v2, and you seem to be using an older version of Stencil. If you can upgrade to the latest version and let me know if this issue still exists, I would appreciate it. We may have a few more fixes for this coming up soon in v2.7.0 as well, so please let us know if that version doesn't improve the behavior. 

Let me know if you need anything else!
